lib/babe/babe.go

```diff
@@ -226,20 +226,23 @@ func (b *Service) EpochLength() uint64 {
 
 // Pause pauses the service ie. halts block production
 func (b *Service) Pause() error {
+	b.Lock()
+	defer b.Unlock()
+
 	if b.paused {
 		return errors.New("service already paused")
 	}
 
-	b.Lock()
-	defer b.Unlock()
-
-	b.pause <- struct{}{}
 	b.paused = true
+	b.pause <- struct{}{}
```

Comment:
nit: seems odd that there are multiple two places in the code where `paused` is set to true.  Also having both `pause` and `paused` seems unnecessary.

Reply:
I'll change the other place to just call `Pause()` instead. it needs the `paused` bool in the `Resume` function to get if the service is paused already or not, is there a better way to do that?

Reply:
ended up removing the second place `paused` is set to true in my other PR

Reply:
You can probably close the pause channel and reinitialize it on resume.
```
func (b *Service) IsPaused() {
	select {
	case: <-b.pause
		return true
	default:
		return false
	}
}
```

```
func (b *Service) Pause() {
	close(b.pause)
}
```

```
func otherFunc() {
	select {
	case <-time.After(time.Until(epochStartTime)):
	case <-b.ctx.Done():
		return nil
	case <-b.pause:
		return nil
	}
}
```